System/Random.hs

```diff
@@ -445,11 +438,18 @@ Minimal complete definition: 'randomR' and 'random'.
 
 -}
 
-class Random a where
-  randomRM :: MonadRandom g m => (a, a) -> g -> m a
 
-  randomM :: MonadRandom g m => g -> m a
+class Uniform a where
+  uniform :: MonadRandom g m => g -> m a
 
+class UniformRange a where
+  uniformR :: MonadRandom g m => (a, a) -> g -> m a
+
+
+{-# DEPRECATED randomR "In favor of `uniformR`" #-}
+{-# DEPRECATED randomRIO "In favor of `uniformR`" #-}
+{-# DEPRECATED randomIO "In favor of `uniformR`" #-}
```

Comment:
Main point of `randomIO`/`randomRIO` is easy access to randomness when you don't care about nor quality nor speed and only want convenience. Wouldn't it better to add IO-based generator for new style API and recommend it? 

```haskell
instance MonadIO m => MonadRandom IOGen m where
  ...
```

Reply:
I think having a global generator is a really bad idea. It is neither fast nor it is deterministic.

Global variables are a major wart in Haskell design and I am not going to promote it.

At the same time an `IOGen` is something that we potentially could do, but that is an orthogonal issue, so we should create a separate issue for that.

Reply:
I opened #15 for this